src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -152,12 +162,85 @@ export function createJourneyApi(
 
       const from = inputToLocation(input, 'from');
       const to = inputToLocation(input, 'to');
-
       const when =
         input.searchTime.mode !== 'now'
           ? new Date(input.searchTime.dateTime)
           : new Date();
 
+      if (input.via) {
+        const via = inputToViaLocation(input);
+        const queryVariables = {
+          from,
+          to,
+          via,
+          when,
+          modes: journeyModes,
+          cursor: input.cursor,
+          ...DEFAULT_JOURNEY_CONFIG,
+        };
+
+        const result = await client.query<
+          ViaTripsQuery,
+          ViaTripsQueryVariables
+        >({
+          query: ViaTripsDocument,
+          variables: queryVariables,
+        });
+
+        if (result.error || result.errors) {
+          throw result.error || result.errors;
+        }
+
+        const { tripPatternCombinations, tripPatternsPerSegment } =
+          result.data.viaTrip;
+
+        // Create list to make all trip pattern combinations more accessable.
+        const tripPatternCombinationList = findTripPatternCombinationsList(
+          tripPatternCombinations,
+        );
+
+        // Find trip patterns from-via and via-to destination.
+        const tripPatternsFromVia = tripPatternsPerSegment[0].tripPatterns;
+        const tripPatternsViaTo = tripPatternsPerSegment[1].tripPatterns;
+
+        // Find all possible trip patterns where the legs from the from-via location and the via-to location are concatinated.
+        const tripPatternsFromViaTo = findTripPatternsFromViaTo(
+          tripPatternCombinationList,
+          tripPatternsFromVia,
+          tripPatternsViaTo,
+        );
+
+        const data: RecursivePartial<TripData> = mapResultToTrips(
+          {
+            nextPageCursor: undefined,
+            previousPageCursor: undefined,
```

Comment:
The "load more trip patterns" function on the assistant page won't work without the cursors. I see that the viaTrip input can take a cursor, but I'm not sure how to retrieve one for the current search. Another approach is to modify the search time when fetching more trip patterns, but that could potentially result in duplicate results and/or missing patterns. Did you look into this?

Reply:
I've looked into option number two a bit by investigating if the search time could be passed as a parameter to the query, instead of the current option where the search time is a stativ value. As you mentioned, this could potentially lead to duplicate results and/or missing patterns. I can also imagine that by increasing the search window, the query will take longer to complete. Of course, the query can be fast enough even with increased search time.

But I´ll take a look if it is possible to retrieve a cursor from the current search.

Reply:
The fetchMore button is now disabled when tripQuery.via is defined. 

<img width="257" alt="Screenshot 2024-02-05 at 11 18 56" src="https://github.com/AtB-AS/planner-web/assets/59939294/8cb48c4a-7cc8-4037-bf8c-ec7d7070b53a">